otherlibs/runtime_events/runtime_events_consumer.c

```diff
@@ -216,21 +217,29 @@ caml_runtime_events_create_cursor(const char_os* runtime_events_path, int pid,
 
   /* This cast is necessary for compatibility with Illumos' non-POSIX
     mmap/munmap */
-  cursor->metadata = (struct runtime_events_metadata_header *)
-                      mmap(NULL, cursor->ring_file_size_bytes, mmap_prot,
+  cursor->map = (void*) mmap(NULL, cursor->ring_file_size_bytes, mmap_prot,
```

Comment:
I don't think the comment and the cast hold anymore, since `mmap` returns a `void *`, even on [illumos](https://illumos.org/man/2/mmap).

Reply:
I can remove it. Do we have Illumos anywhere in the CI to double check? (maybe it referred to an old Illumos version)

Reply:
See https://github.com/ocaml/ocaml/pull/11299 for context on when this was added. It looks like Illumos has both POSIX compliant and non-compliant headers, but to get the POSIX compliant ones some additional defines are needed, which need changes elsewhere.
This may get solved if https://github.com/ocaml/ocaml/pull/10505 is progressed.

For now I'd keep the cast, it can be dropped later.